api4/file.go

```diff
@@ -298,16 +294,39 @@ func getPublicFile(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func writeFileResponse(filename string, contentType string, bytes []byte, toDownload bool, w http.ResponseWriter, r *http.Request) *model.AppError {
+func writeFileResponse(filename string, contentType string, bytes []byte, forceDownload bool, w http.ResponseWriter, r *http.Request) *model.AppError {
 	w.Header().Set("Cache-Control", "max-age=2592000, private")
 	w.Header().Set("Content-Length", strconv.Itoa(len(bytes)))
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+
+	if contentType == "application/javascript" || contentType == "application/ecmascript" ||
+		contentType == "text/javaScript" || contentType == "text/ecmascript" ||
+		contentType == "application/x-javascript" || contentType == "text/html" {
+		contentType = "application/octet-stream"
+	}
 
 	if contentType != "" {
 		w.Header().Set("Content-Type", contentType)
 	} else {
 		w.Header().Del("Content-Type") // Content-Type will be set automatically by the http writer
```

Comment:
this case worries me a _little_ bit. i _think_ it's fine, i'm not 100% comfortable trusting the automatic content-type to not end up setting this to one of our blacklisted types. i'd feel a little better if we set this explicitly in all cases (i think `w.Header().Set("Content-Type", "application/octet-stream")` would be fine)